pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -190,6 +211,35 @@ public static List<NativeQuery> parseJdbcSql(String query, boolean standardConfo
           }
           break;
 
+        case ':': // possibly named placerholder start'
+          if (processParameters && PlaceholderStyles.NAMED.placeholderStyleIsAccepted(allowedPlaceholderStyles) && currentCommandType.supportsParameters()) {
+
+            nativeSql.append(aChars, fragmentStart, i - fragmentStart);
+            int end = Parser.parseNamedPlaceholder(aChars, i);
+            if (end != i) {
+              fragmentStart = i;
+
+              // Skip the colon from the captured name.
+              final String bindName = query.substring(fragmentStart + 1, end + 1);
+              final int bindIndex = paramCtx.addNamedParameter(nativeSql.length(), ParameterContext.BindStyle.NAMED, ":", bindName);
```

Comment:
What if we keep the full token image? I might be missing something, however, it looks like the split into `prefix` and `name` adds little value, and sometime later there's `":".concat(..`

Reply:
Just in case, `setAbc(String, ...)` seem to use the full name (e.g.  `:name`, `$5`), so the split does not seem to be required for it.